### PR TITLE
management: fix runtime-edit crash in py3

### PR DESCRIPTION
**Pull request by Sasasu**
fllow up https://github.com/greenplum-db/plcontainer/pull/665

```
Traceback (most recent call last):
  File "/home/sa/GPDB/plcontainer/./management/bin/plcontainer", line 974, in main
    changed = edit_file(tmp_file, options.editor)
  File "/home/sa/GPDB/plcontainer/./management/bin/plcontainer", line 647, in edit_file
    h1 = get_file_md5(filename)
  File "/home/sa/GPDB/plcontainer/./management/bin/plcontainer", line 486, in get_file_md5
    h.update(l)
TypeError: Strings must be encoded before hashing
```

### CODE REVIEW

File: management/bin/plcontainer [hunk 1]
```diff
@@ -483,7 +483,7 @@ def get_file_md5(file):
     h = hashlib.md5()
     f = open(file, 'r')
     for l in f:
-        h.update(l)
+        h.update(l.encode('utf-8'))
```
Comment (higuoxing) [inline management/bin/plcontainer:486]: Seems that we can simplify it to:

```
def get_file_md5(file):
    with open(file, 'rb') as f:
        return hashlib.md5(f.read()).digest()
```

> hash.update(data)
> Update the hash object with the [bytes-like object](https://docs.python.org/3/glossary.html#term-bytes-like-object). Repeated calls are equivalent to a single call with the concatenation of all the arguments: m.update(a); m.update(b) is equivalent to m.update(a+b).

Ref: https://docs.python.org/3/library/hashlib.html#hashlib.hash.update

Comment (Sasasu) [reply management/bin/plcontainer:486]: from py2.7: `Update the hash object with the string arg.`
I think this does not support python2

Comment (Sasasu) [reply management/bin/plcontainer:486]: 401ecfe5

**Review by higuoxing — APPROVED** [review]

**Comment by yihong0618** [comment]
nice rewitre, and I think all the open close things can replace to `with open() as f` in this file

**Comment by Sasasu** [comment]
test in both gp6 with py2 and gp7 with py3

**Pull request merged**